### PR TITLE
Expand the guide for native test

### PR DESCRIPTION
**Pull request by Thevakumar-Luheerathan**
## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

**Review by shafreenAnfar — APPROVED** [review]

File: swan-lake/learn-the-platform/native-support/build-a-native-executable.md [hunk 1]
```diff
@@ -351,7 +351,52 @@ After the environment is set up, follow the steps below to build the native exec
    Hello, Docker!
    ```
    
+## Execute test with the native image
```
Comment (praneesha) [inline swan-lake/learn-the-platform/native-support/build-a-native-executable.md:354]:
```suggestion
## Execute tests with the native image
```

File: swan-lake/learn-the-platform/native-support/build-a-native-executable.md [hunk 2]
```diff
@@ -351,7 +351,52 @@ After the environment is set up, follow the steps below to build the native exec
...
+
+It is recommended to use the JVM for verifying the functionality of the application with sufficient tests and code coverage as running tests against the native image could take time. Running tests against the native image is only required when using libraries that are not verified against GraalVM native image build. All Ballerina standard libraries are verified and guaranteed to produce native image builds without any issues. The complete list of verified libraries can be found at the top of the article.
```
Comment (praneesha) [inline swan-lake/learn-the-platform/native-support/build-a-native-executable.md:356]:
```suggestion
It is recommended to use the JVM for verifying the functionality of the application with sufficient tests and code coverage as running tests against the native image could take time. Running tests against the native image is only required when using libraries that are not verified against the GraalVM native image build. All Ballerina standard libraries are verified and guaranteed to produce native image builds without any issues. The complete list of verified libraries can be found at the top of the article. 
```
Need to give the exact link without saying `at the top of the article`.

File: swan-lake/learn-the-platform/native-support/build-a-native-executable.md [hunk 3]
```diff
@@ -351,7 +351,52 @@ After the environment is set up, follow the steps below to build the native exec
    Hello, Docker!
    ```
    
+## Execute test with the native image
+
+It is recommended to use the JVM for verifying the functionality of the application with sufficient tests and code coverage as running tests against the native image could take time. Running tests against the native image is only required when using libraries that are not verified against GraalVM native image build. All Ballerina standard libraries are verified and guaranteed to produce native image builds without any issues. The complete list of verified libraries can be found at the top of the article. 
+
+Therefore, if the application is depending on a package that is not verified against GraalVM native build, it is recommended to run the tests against the native image to ensure there are no runtime errors. `bal test --native` can be used for this purpose.
```
Comment (praneesha) [inline swan-lake/learn-the-platform/native-support/build-a-native-executable.md:358]:
```suggestion
Therefore, if the application is depending on a package that is not verified against the GraalVM native build, it is recommended to run the tests against the native image to ensure that there are no runtime errors by executing the `bal test --native` command.
```

File: swan-lake/learn-the-platform/native-support/build-a-native-executable.md [hunk 4]
```diff
@@ -351,7 +351,52 @@ After the environment is set up, follow the steps below to build the native exec
...
+
+Also, native testing can be scheduled as a daily check within CI pipelines to maintain compatibility with GraalVM.
```
Comment (praneesha) [inline swan-lake/learn-the-platform/native-support/build-a-native-executable.md:360]:
```suggestion
Also, native testing can be scheduled as a daily check within CI pipelines to maintain compatibility with GraalVM.
```
What is `CI`?

File: swan-lake/learn-the-platform/native-support/build-a-native-executable.md [hunk 5]
```diff
@@ -351,7 +351,52 @@ After the environment is set up, follow the steps below to build the native exec
...
+
+Follow the steps below to run the tests with the native image.
+
+
+1. Follow the steps 1 and 2 in [Build a native executable](#ballerina-native-image).
+
+2. Replace the content of the `service_test.bal` file with the following under the `tests` folder.
```
Comment (praneesha) [inline swan-lake/learn-the-platform/native-support/build-a-native-executable.md:367]:
```suggestion
2. Replace the content of the `service_test.bal` file in the `tests` folder with the following.  
```

File: swan-lake/learn-the-platform/native-support/build-a-native-executable.md [hunk 6]
```diff
@@ -351,7 +351,52 @@ After the environment is set up, follow the steps below to build the native exec
...
+   @test:Config {}
+   function testServiceWithProperName() {
+      string|error response = testClient->get("/greeting");
+      test:assertEquals(response, "Hello, World!");
+   }
+
+   ```
+
+3.  Run `bal test --native` to run the tests using the native executable. 
+      > **Info:**  This command will build a native executable with tests similar to bal build --native , and the tests will be executed by running this native executable.
```
Comment (praneesha) [inline swan-lake/learn-the-platform/native-support/build-a-native-executable.md:383]:
```suggestion
      > **Info:**  This command will build a native executable with tests similar to `bal build --native`, and the tests will be executed by running this native executable.
```

File: swan-lake/learn-the-platform/native-support/build-a-native-executable.md [hunk 7]
```diff
@@ -351,7 +351,52 @@ After the environment is set up, follow the steps below to build the native exec
...
+      ```
+               Running Tests
+               hello_world
+                        [pass] testServiceWithProperName
+                        1 passing
+                        0 failing
+                        0 skipped
+      ```
+
+You have now tested a simple Ballerina HTTP server application for GraalVM compatibility.
```
Comment (praneesha) [inline swan-lake/learn-the-platform/native-support/build-a-native-executable.md:393]:
```suggestion
Now, you tested a simple Ballerina HTTP server application for GraalVM compatibility.
```

File: swan-lake/learn-the-platform/native-support/build-a-native-executable.md (hunk 4 above)
Comment (Thevakumar-Luheerathan) [reply swan-lake/learn-the-platform/native-support/build-a-native-executable.md:360]: It's continuous integration.

**Review by praneesha — APPROVED** [review]

**Pull request merged**